docarray/helper.py

```diff
@@ -321,3 +322,69 @@ def dataclass_from_dict(klass, dikt):
         if isinstance(dikt, (tuple, list)):
             return [dataclass_from_dict(klass.__args__[0], f) for f in dikt]
         return dikt
+
+
+def protocol_and_compress_from_file_path(
+    file_path: str,
+    default_protocol: Optional[str] = None,
+    default_compress: Optional[str] = None,
+) -> Tuple[Optional[str], Optional[str]]:
+    """Extract protocol and compression algorithm from a string, use defaults if not found.
+
+    :param file_path: path of a file.
+    :param default_protocol: default serialization protocol used in case not found.
+    :param default_compress: default compression method used in case not found.
+
+    Examples:
+
+    >>> protocol_and_compress_from_file_path('./docarray_fashion_mnist.protobuff.gzip')
+    ('protobuf', 'gzip')
+
+    >>> protocol_and_compress_from_file_path('/Documents/docarray_fashion_mnist.protobuff')
+    ('protobuf', None)
+
+    >>> protocol_and_compress_from_file_path('/Documents/docarray_fashion_mnist.gzip')
+    (None, gzip)
+    """
+
+    ALLOWED_PROTOCOLS = {'pickle', 'protobuf', 'protobuf-array', 'pickle-array'}
```

Comment:
I think this should be defined somewhere more globally to make sure is better mantainable?

Reply:
Agree, maybe in `docarray/array/mixins/io/binary.py`